tests/integration_tests/utils.py

```diff
@@ -758,3 +759,273 @@ def filter(stats):
     finally:
         # Remove results/intermediate data saved to disk
         shutil.rmtree(output_dir, ignore_errors=True)
+
+
+class ParameterUpdateError(Exception):
+    pass
```

Comment:
Could you move the changes in this file to a separate library, i.e. `tests/integration_tests/parameter_update_utils.py`?

This utils file is a bit large, and it could be good to implement a bit more separation.

Reply:
Create separate module per review comment